Sources/SwiftProcessing/Core/Sketch3D/Sketch3DPrimitives.swift

```diff
@@ -4,110 +4,116 @@ import SceneKit
 public extension Sketch {
 
     func shapeCreate(_ tag: String, _ geometry: SCNGeometry,_ type: String) {
-        
-        let newtag = tag + "r" + self.settings.fill.red.description + "g" + self.settings.fill.green.description + "b" + self.settings.fill.blue.description + "a" + self.settings.fill.alpha.description
-        
+
+        var newtag = tag + "r" + self.settings.fill.red.description + "g" + self.settings.fill.green.description + "b" + self.settings.fill.blue.description + "a" + self.settings.fill.alpha.description
+
         if var shapeNode = self.currentTransformationNode.getAvailableShape(newtag) {
-            
-    
+
+
         } else {
-            
+
             geometry.firstMaterial?.diffuse.contents = UIColor(red: self.settings.fill.red/255.0, green: self.settings.fill.green/255.0, blue: self.settings.fill.blue/255.0, alpha: self.settings.fill.alpha)
-            
+
+            if self.texture != nil && self.textureEnabled {
+                geometry.firstMaterial?.diffuse.contents = self.texture!.currentFrame()
```

Comment:
not a request for a change, but I wonder what it'd take to support a GIF as a texture 🤣